server/src/test/java/org/elasticsearch/action/admin/indices/alias/GetAliasesResponseTests.java

```diff
@@ -0,0 +1,249 @@
+                "  \"aliases\": {\n" +
+                "   \"alias\": {}\n" +
+                "  }\n" +
+                " }\n" +
+                "}";
+        XContentParser parser = createParser(JsonXContent.jsonXContent, xContent);
+        GetAliasesResponse response = GetAliasesResponse.fromXContent(parser);
+        assertThat(response.status(), equalTo(RestStatus.NOT_FOUND));
+        assertThat(response.errorMsg(), equalTo("alias [something] missing"));
+        assertThat(response.getAliases().size(), equalTo(1));
+        assertThat(response.getAliases().get("index").size(), equalTo(1));
+        assertThat(response.getAliases().get("index").get(0), notNullValue());
+        assertThat(response.getAliases().get("index").get(0).alias(), equalTo("alias"));
+    }
+
+    public void testFromXContentWithElasticsearchException() throws IOException {
+        String xContent =
+                "{\n" +
+                "  \"error\": {\n" +
+                "    \"root_cause\": [\n" +
+                "      {\n" +
+                "        \"type\": \"index_not_found_exception\",\n" +
+                "        \"reason\": \"no such index\",\n" +
+                "        \"resource.type\": \"index_or_alias\",\n" +
+                "        \"resource.id\": \"index\",\n" +
+                "        \"index_uuid\": \"_na_\",\n" +
+                "        \"index\": \"index\"\n" +
+                "      }\n" +
+                "    ],\n" +
+                "    \"type\": \"index_not_found_exception\",\n" +
+                "    \"reason\": \"no such index\",\n" +
+                "    \"resource.type\": \"index_or_alias\",\n" +
+                "    \"resource.id\": \"index\",\n" +
+                "    \"index_uuid\": \"_na_\",\n" +
+                "    \"index\": \"index\"\n" +
+                "  },\n" +
+                "  \"status\": 404\n" +
+                "}";
+
+        XContentParser parser = createParser(JsonXContent.jsonXContent, xContent);
+        ElasticsearchException expectThrows = expectThrows(ElasticsearchException.class, () -> GetAliasesResponse.fromXContent(parser));
+        assertThat(expectThrows.status(), equalTo(RestStatus.NOT_FOUND));
+        assertThat(expectThrows.getMessage(), equalTo("Elasticsearch exception [type=index_not_found_exception, reason=no such index]"));
+    }
+
+    public void testFromXContentWithNoAliasFound() throws IOException {
+        String xContent =
+                "{\n" +
+                "  \"error\": \"alias [aa] missing\",\n" +
+                "  \"status\": 404\n" +
+                "}";
+        XContentParser parser = createParser(JsonXContent.jsonXContent, xContent);
+        ElasticsearchException expectThrows = expectThrows(ElasticsearchException.class, () -> GetAliasesResponse.fromXContent(parser));
+        assertThat(expectThrows.status(), equalTo(RestStatus.NOT_FOUND));
+        assertThat(expectThrows.getMessage(), equalTo("alias [aa] missing"));
+    }
+
```

Comment:
I think that we should also have some serialization test that cover the bw comp changes we are making.